cogito/putter.go

```diff
@@ -262,7 +262,11 @@ func checkGitRepoDir(dir, owner, repo string) error {
 	if err != nil {
 		return fmt.Errorf(".git/config: remote: %w", err)
 	}
-	left := []string{"github.com", owner, repo}
```

Comment:
I do think that its extremely important to test and verify all 3 components.
Input from Concourse git resource can come from anywhere (gitlab, bitbucket, github enterprise, etc...) so just comparing with hardcoded github.com is wrong.

Reply:
Maybe this: https://github.com/google/go-github/blob/56ca0e5d559f691400370d2341542f06c0305df9/github/github.go#L350

will give us more insight into behavior of GitHub Enterprise...

Reply:
> Input from Concourse git resource can come from anywhere (gitlab, bitbucket, github enterprise, etc...)

True, but I want to clarify something: cogito will work only in 2 cases:
- input comes from github.com. In this case we can validate before hitting the GH API.
- input comes from a github enterprise installation. In this case, there is NO way for us to actually know that it comes from a GH enterprise as opposed to, for example, from codeberg (https://codeberg.org/explore/repos).

This brings me back a bit to my feeling that we have to abandon validation before hitting the gh API.

On the other hand, I need to take some time to think it through, currently I have only a feeling and I might be wrong for sure. Could you please ping me when I am back? Merci.